nomad/structs/structs.go

```diff
@@ -10904,6 +10951,23 @@ func (p *Plan) AppendPreemptedAlloc(alloc *Allocation, preemptingAllocID string)
 	p.NodePreemptions[node] = append(existing, newAlloc)
 }
 
+// AppendUnknownAlloc marks an allocation as unknown.
+func (p *Plan) AppendUnknownAlloc(alloc *Allocation) {
+	// TODO (derek): review this with Tim
+
+	// Already created a copy in reconiler
```

Comment:
Needs review.

Reply:
I'm not sure if I need to set the `Job` and `Resources` `nil` here. I modeled this after `AppendStoppedAlloc`, but I don't understand why these are set to `nil`.

Reply:
Thanks for the guidance on where to look. It seems like this is an attempt to reduce the payload size as it serializes a `ApplyPlanResultRequest`. There is one Job on the whole request, and the `NodeAllocations` are all added, so stripping these fields should significantly reduce the size.

Reply:
👍